client/src/components/bookings/RequestStatus.js

```diff
@@ -35,13 +40,31 @@ function RequestStatus({ request, modeTime }) {
   const classes = useStyles();
   const [acceptButtonSubmitting, setAcceptButtonSubmitting] = useState(false);
   const [declineButtonSubmitting, setDeclineButtonSubmitting] = useState(false);
+  const [cards, setCards] = useState(null);
+  const [paymentModal, togglePaymentModal] = useState(false);
 
   // Get dispatch method from context
   const dispatch = useContext(AuthDispatchContext);
+  const { errors } = useContext(AuthStateContext);
+
+  useEffect(() => {
+    if (errors.length > 0) {
+      setAcceptButtonSubmitting(false);
+    }
+  }, [errors.length]);
 
   return (
     <Box className={classes.lightGreyColor}>
-      {/* Show Accept/Decline Buttons if:
+      {paymentModal && (
+        <PaymentModal
+          togglePaymentModal={togglePaymentModal}
+          request={request}
+          cards={cards}
+          setCards={setCards}
+        />
+      )}
+      {/* 
```

Comment:
I think this large block comment is not necessary. I think if you save your conditional to variables with names like `showAcceptDecline` and use those to conditionally render it will convey the same information without having this large of a comment.

Generally large comments explaining functionality are not ideal, it's usually better to rewrite your code to be clearer.

Reply:
Updated. The conditionals are in variables now.